llvm/lib/Target/AArch64/AsmParser/AArch64AsmParser.cpp

```diff
@@ -8100,54 +8165,8 @@ bool AArch64AsmParser::parsePrimaryExpr(const MCExpr *&Res, SMLoc &EndLoc) {
 bool AArch64AsmParser::parseAuthExpr(const MCExpr *&Res, SMLoc &EndLoc) {
   MCAsmParser &Parser = getParser();
   MCContext &Ctx = getContext();
-
   AsmToken Tok = Parser.getTok();
 
-  // Look for '_sym@AUTH' ...
```

Comment:
As I understand it, this can be stripped out as unecessary as the parsing is limited to data directives?

Reply:
Right. In the absence of `@` specifiers, the expression can be handled by the generic `getParser().parseExpression(Res)`.